src/realm/array_mixed.cpp

```diff
@@ -319,6 +321,90 @@ void ArrayMixed::verify() const
     // TODO: Implement
 }
 
+ref_type ArrayMixed::typed_write(ref_type top_ref, _impl::ArrayWriterBase& out, Allocator& alloc)
+{
+    if (out.only_modified && alloc.is_read_only(top_ref))
+        return top_ref;
+
+    ArrayRef top(alloc);
+    top.init_from_ref(top_ref);
+    size_t sz = top.size();
+    TempArray written_leaf(sz);
+
+    /*
+     Mixed stores things using different arrays. We need to take into account this in order to
+     understand what we need to compress and what we can instead leave not compressed.
+
+     The main subarrays are:
+
+     composite array : index 0
+     int array : index 1
+     pair_int array: index 2
+     string array: index 3
+     ref array: index 4
+     key array: index 5
+
+     Description of each array:
+     1. composite array: the data stored here is either a small int (< 32 bits) or an offset to one of
+        the other arrays where the actual data is.
+     2. int and pair int arrays, they are used for storing integers, timestamps, floats, doubles,
+     decimals, links. In general we can compress them, but we need to be careful, controlling the col_type
+     should prevent compressing data that we want to leave in the current format.
+     3. string array is for strings and binary data (no compression for now)
+     4. ref array is actually storing refs to collections. they can only be BPlusTree<int, Mixed> or
+     BPlusTree<string, Mixed>.
+     5. key array stores unique identifiers for collections in mixed (integers that can be compressed)
+     */
+    Array composite(alloc);
+    composite.init_from_ref(top.get_as_ref(0));
+    written_leaf.set_as_ref(0, composite.write(out, true, out.only_modified, false));
+    for (size_t i = 1; i < sz; ++i) {
+        ref_type new_ref = 0;
+        if (auto ref = top.get(i)) {
+            if (i < 3) { // int, and pair_int
```

Comment:
@jedelbo can you disable compression for the composite array? I wanted to do it in my branch but, since you are touching this code, we can do it here. We want to compress only `if i >=1 and i < 3` and leave the refs in the composite array uncompressed (this has benefits on many queries)

Reply:
Composite array is written outside this loop with compression set to false.

Reply:
ah, sorry my bad, I did not notice this:
`for (size_t i = 1; i < sz; ++i) ` ... thanks!